horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/ServiceMediator.java

```diff
@@ -222,4 +236,27 @@ public void validateDataset(Integer datasetId) {
     public void validateSchema(int schemaId) {
         schemaService.revalidateAll(schemaId);
     }
+
+    public <T> void publishEvent(AsyncEventChannels channel, int testId, T payload) {
+        if (testMode ) {
+            log.debugf("Publishing test %d on %s: %s", testId, channel, payload);
+//        eventBus.publish(channel.name(), new MessageBus.Message(BigInteger.ZERO.longValue(), testId, 0, payload));
+            events.putIfAbsent(channel, new HashMap<>());
+            BlockingQueue<Object> queue = events.get(channel).computeIfAbsent(testId, k -> new LinkedBlockingQueue<>());
+            queue.add(payload);
+        } else {
+            //no-op
+        }
+    }
+
+    public <T> BlockingQueue<T> getEventQueue(AsyncEventChannels channel, Integer id) {
+        if (testMode) {
+            events.putIfAbsent(channel, new HashMap<>());
+            BlockingQueue<?> queue = events.get(channel).computeIfAbsent(id, k -> new LinkedBlockingQueue<>());
+            return (BlockingQueue<T>) queue;
+        } else {
+            return null;
+        }
+    }
+
```

Comment:
I am not that sure we want to include these changes as they were not part of the original PR, this looks like a conflict resolution.

Reply:
Removed un-wanted changes, CI is green again :heavy_check_mark: